src/Mvc/Mvc.Core/src/Routing/DynamicControllerEndpointMatcherPolicy.cs

```diff
@@ -42,8 +43,13 @@ public bool AppliesToEndpoints(IReadOnlyList<Endpoint> endpoints)
 
             for (var i = 0; i < endpoints.Count; i++)
             {
-                var metadata = endpoints[i].Metadata.GetMetadata<DynamicControllerMetadata>();
-                if (metadata != null)
+                if (endpoints[i].Metadata.GetMetadata<DynamicControllerMetadata>() != null)
+                {
+                    // Found a dynamic controller endpoint
+                    return true;
+                }
+
+                if (endpoints[i].Metadata.GetMetadata<DynamicControllerRouteValueTransformerMetadata>() != null)
```

Comment:
The code is like 15% more complicated because we handle two cases in a single policy. The upside is that this class has to maintain a cache of computed data per-endpoint. I don't want to duplicate that cache and use 2x the memory.